src/vos/vos_ilog.c

```diff
@@ -154,7 +154,8 @@ vos_parse_ilog(struct vos_ilog_info *info, daos_epoch_t epoch,
 		}
 
 		entry_epc = entry.ie_id.id_epoch;
-		if (entry_epc > epoch) {
+		if (entry_epc > epr->epr_hi) {
+			info->ii_full_scan = false;
```

Comment:
Minor: When the intention of this call is for "punched check", maybe we'd return immediately as soon as we detect "not punched"?

Reply:
fixed